tools/install-monorepo.sh

```diff
@@ -15,143 +17,344 @@ function usage {
+			# Effectively only support Debian.
+			die "Installer script requires 'apt' to ensure build-essential package is installed."
+		fi
 
-if ! command -v curl &>/dev/null; then
-	die "Installer script requires 'curl' to be installed"
-fi
+		sudo apt update
+		if [[ $(dpkg-query -W --showformat='${db:Status-Status}' build-essential 2>/dev/null) != 'installed' ]]; then
+			echo "build-essential: not found"
+			echo "Installing build-essential..."
+			sudo apt install build-essential || die "Unable to install build-essential!"
+			echo "Installed build-essential."
+		fi
+		echo "build-essential: available"
+	fi
 
-# Check of Homebrew and nvm are installed
-info "Checking if Homebrew is installed..."
-if ! command -v brew &>/dev/null; then
-	info "Installing Homebrew"
-	/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
-	if [[ -n "$ON_LINUX" ]]; then # Add homebrew to PATH
-		echo 'eval "$("/home/linuxbrew/.linuxbrew/bin/brew" shellenv)"' >> "$HOME/.profile"
-		eval "$("/home/linuxbrew/.linuxbrew/bin/brew" shellenv)"
-		PATH="/home/linuxbrew/.linuxbrew/bin:$PATH"
+	# Check for curl and git
+	if ! has_command git; then
+		echo "git: not found"
+		if [[ "$OS" == "Linux" ]]; then
+			echo "Installing git..."
+			sudo apt install git || die "Unable to install git!"
+			echo "Installed git."
+		else
+			die "Installer script requires git to be installed."
+		fi
 	fi
-else
-	info "Updating brew"
-	brew update
-	# Brew can be finicky on MacOS
-	if [[ $? -ne 0 ]]; then
-		 echo "Reinstalling Homebrew"
-		 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+	echo "git: available"
+
+	if ! has_command curl; then
+		echo "curl: not found"
+		if [[ "$OS" == "Linux" ]]; then
+			echo "Installing curl..."
+			sudo apt install curl || die "Unable to install curl!"
+			echo "Installed curl."
+		else
+			die "Installer script requires curl to be installed."
+		fi
 	fi
-fi
+	echo "curl: available"
+}
 
-info "Checking if NVM is installed..."
-if ! command -v nvm &>/dev/null; then
-	info "Installing nvm"
-	curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash && export NVM_DIR=$HOME/.nvm && source $NVM_DIR/nvm.sh  --no-use
-else
-	info "Updating nvm"
-	curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.39.0/install.sh | bash
-fi
+# Checks for, installs, and updates Homebrew.
+function do_homebrew {
+	if ! has_command brew; then
+		echo "brew: not found"
+		echo "Installing brew..."
+		/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" || die "Unable to install brew!"
+		if [[ "$OS" == "Linux" ]]; then
+			HOMEBREW_PREFIX="/home/linuxbrew/.linuxbrew"
+		else
+			HOMEBREW_PREFIX="/opt/homebrew"
+		fi
 
-# Install and use the correct version of Node.js
-info "Installing Node.js"
-nvm install && nvm use
+		# Add brew env to current script for use later.
+		eval "\$(${HOMEBREW_PREFIX}/bin/brew shellenv)"
```

Comment:
This doesn't work right, at least in bash.
```suggestion
		eval "$(${HOMEBREW_PREFIX}/bin/brew shellenv)"
```

Reply:
Yeah, copy/paste mistake. 😆